packages/components/src/Button/Button.tsx

```diff
@@ -1,149 +1,46 @@
-  ): void;
-}
-
-interface ButtonIconProps extends ButtonFoundationProps {
-  readonly icon: IconNames;
-  readonly ariaLabel: string;
-}
-
-interface ButtonLabelProps extends ButtonFoundationProps {
-  readonly label: string;
-}
-
-interface ButtonAnchorProps extends ButtonFoundationProps {
-  /**
-   * Used to create an 'href' on an anchor tag.
-   */
-  readonly url?: string;
-}
-
-interface ButtonLinkProps<S = unknown> extends ButtonFoundationProps {
-  /**
-   * **Deprecated**: to will be removed in the next major version
-   * @deprecated
-   */
-  readonly to?: LinkProps<S>["to"];
-}
-
-interface BaseActionProps extends ButtonFoundationProps {
-  readonly variation?: "work" | "learning" | "subtle" | "destructive";
-  readonly type?: "primary" | "secondary" | "tertiary";
-}
-
-interface DestructiveActionProps extends ButtonFoundationProps {
-  readonly variation: "destructive";
-  readonly type?: "primary" | "secondary" | "tertiary";
-}
-
-interface SubmitActionProps
-  extends Omit<BaseActionProps, "external" | "onClick"> {
-  readonly name?: string;
-  readonly submit: boolean;
-  readonly type?: "primary" | "secondary" | "tertiary";
-  readonly value?: string;
-}
-
-interface SubmitButtonProps
-  extends Omit<ButtonFoundationProps, "external" | "onClick"> {
-  /**
-   * Allows the button to submit a form
-   */
-  submit: boolean;
-}
-
-interface BasicButtonProps extends ButtonFoundationProps {
-  /**
-   * Used to override the default button role.
-   */
-  readonly role?: string;
+  return (
+    <Button.Provider size={size}>
+      <ButtonWrapper {...props} />
+    </Button.Provider>
+  );
 }
 
-export type ButtonProps = XOR<
-  BaseActionProps,
-  XOR<DestructiveActionProps, SubmitActionProps>
-> &
-  XOR<
-    XOR<SubmitButtonProps, BasicButtonProps>,
-    XOR<ButtonLinkProps, ButtonAnchorProps>
-  > &
-  XOR<ButtonIconProps, ButtonLabelProps>;
-
-export function Button(props: ButtonProps) {
+function ButtonWrapper(props: ButtonProps) {
   const {
     ariaControls,
     ariaHaspopup,
     ariaExpanded,
     ariaLabel,
     disabled = false,
     external,
-    fullWidth,
-    icon,
-    label,
-    iconOnRight,
     id,
-    loading,
     name,
     onClick,
     onMouseDown,
     role,
-    size = "base",
-    type = "primary",
-    url,
-    to,
     value,
-    variation = "work",
     submit,
+    to,
+    url,
   } = props;
 
-  const buttonClassNames = classnames(styles.button, styles[size], {
-    [styles.onlyIcon]: icon && !label,
-    [styles.hasIconAndLabel]: icon && label,
-    [styles.iconOnRight]: iconOnRight,
-    [styles[variation]]: variation,
-    [styles[type]]: type,
-    [styles.disabled]: disabled,
-    [styles.fullWidth]: fullWidth,
-    [styles.loading]: loading,
-  });
+  const { combined } = useButtonStyles(props);
 
-  const buttonType: ButtonType = submit ? "submit" : "button";
+  const buttonType: HTMLButtonType = submit ? "submit" : "button";
```

Comment:
Any way to support reset? Preferably just leverage the type prop on buttons so that we get freebies when used in forms?